src/tactic/ui/tools/repo_browser_wdg.py

```diff
@@ -2017,43 +2006,54 @@ def get_dir_context_menu(my, mode="strict"):
                 'type': 'click_up',
                 'cbjs_action': '''
                 var activator = spt.smenu.get_activator(bvr);
+                
                 var relative_dir = activator.getAttribute("spt_relative_dir");
                 var original_dir = activator.getAttribute("spt_reldir");
-
                 var original_el = activator.getElement(".spt_dir_value");
-                original_el.setStyle("display", "none");
-                
-                // Inject a input
-                var input = $(document.createElement("input"));
-                input.value = original_dir;
-                input.setAttribute("type", "text");
-                input.setStyle("width", "200px");
-                input.inject(original_el, "after");
                 
+                // Create dummy entry
+                var div = document.createElement("div");
+                div = $(div);
+                div.setStyles(activator.style);
+                var arrow = "/context/icons/silk/_spt_bullet_arrow_down_dark.png";
+                var icon = "/context/icons/silk/folder.png";
+                var html = '<img src="'+arrow+'"/>';
+                html += '<img src="'+icon+'"/>';
+                html += '<input class="rename_folder_input" type="text" style="z-index:10;"/>';
+                div.innerHTML = html;
+                var input = div.getElement(".rename_folder_input");
+
                 var parts = relative_dir.split("/");
-                input.value = parts[parts.length-1];
+                var original_value = parts[parts.length-1];
+                input.value = original_value;
                 var base_relative_dir = parts.slice(0, parts.length-1).join("/");
                 
+                activator.setStyle("display", "none");
+                div.inject(activator, "after");
+                
                 spt.repo_browser.set_lock(true, "context_menu_action");
+ 
+                var exit = function() {
```

Comment:
shouldn't exit() take 2 args activator and div? js allows this implied variable but is it good practice in this situation?